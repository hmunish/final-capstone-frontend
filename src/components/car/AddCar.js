import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { addCar } from "../../redux/cars/carsSlice";

const AddCar = () => {
  const { isLoading, isCreated, isCreatedError } = useSelector(
    (state) => state.cars,
  );
  const { userId } = useSelector((state) => state.login);
  const dispatch = useDispatch();

  const handleAddCar = (e) => {
    e.preventDefault();
    const name = e.target.name.value;
    const image = e.target.image.value;
    const description = e.target.description.value;
    const deposit = e.target.deposit.value;
    const financeFee = e.target.financeFee.value;
    const optionToPurchaseFee = e.target.optionToPurchaseFee.value;
    const totalAmountPayable = e.target.totalAmountPayable.value;
    const duration = e.target.duration.value;

    dispatch(
      addCar({
        userId,
        name,
        image,
        description,
        deposit,
        financeFee,
        optionToPurchaseFee,
        totalAmountPayable,
        duration,
      }),
    );
  };

  if (isLoading) {
    return (
      <div>
        <h1>Loading...</h1>
      </div>
    );
  }

  return (
    <section className="add-new-car">
      <div className="green-overlay" />
      <form className="add-car-form" onSubmit={(e) => handleAddCar(e)}>
        <h1 className="delete-title">Add a New Car</h1>
        <ul className="form-ul">
          <li className="form-li">
            <label htmlFor="name">
              Name:
              <br />
              <input
                type="text"
                id="name"
                maxLength={30}
                name="name"
                required
                placeholder="Car Name"
              />
            </label>
          </li>
          <li className="form-li">
            <label htmlFor="image">
              Image:
              <br />
              <input type="text" id="image" name="image" required />
            </label>
          </li>
          <li className="form-li">
            <label htmlFor="description">
              Description:
              <br />
              <input
                type="text"
                id="description"
                name="description"
                required
                placeholder="Model"
              />
            </label>
          </li>
          <li className="form-li">
            <label htmlFor="deposit">
              Deposit:
              <br />
              <input
                type="number"
                id="deposit"
                name="deposit"
                required
                placeholder="Deposit"
                min="0"
              />
            </label>
          </li>
          <li className="form-li">
            <label htmlFor="financeFee">
              Finance Fee:
              <br />
              <input
                type="number"
                id="financeFee"
                name="financeFee"
                required
                placeholder="Finance Fee"
                min="0"
              />
            </label>
          </li>
          <li className="form-li">
            <label htmlFor="optionToPurchaseFee">
              Option to Purchase Fee:
              <br />
              <input
                type="number"
                id="optionToPurchaseFee"
                name="optionToPurchaseFee"
                required
                placeholder="Option to Purchase Fee"
                min="0"
              />
            </label>
          </li>
          <li className="form-li">
            <label htmlFor="totalAmountPayable">
              Total Amount Payable:
              <br />
              <input
                type="number"
                id="totalAmountPayable"
                name="totalAmountPayable"
                required
                placeholder="Total Amount Payable"
                min="0"
              />
            </label>
          </li>
          <li className="form-li">
            <label htmlFor="Duration">
              Duration:
              <br />
              <input
                type="number"
                id="duration"
                name="duration"
                required
                placeholder="Duration (Months)"
                min="0"
              />
            </label>
          </li>
          <input type="submit" value="Add Car" />
        </ul>
        {isCreated && <p>Car added successfully</p>}
        {isCreatedError && <p>{isCreatedError}</p>}
      </form>
    </section>
  );
};

export default AddCar;
