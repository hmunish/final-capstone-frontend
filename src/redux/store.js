import { configureStore } from '@reduxjs/toolkit';
import loginReducer from './login/loginSlice';
import signupReducer from './signup/signupSlice';

const store = configureStore({
  reducer: {
    login: loginReducer,
    signup: signupReducer,
  },
});

export default store;
