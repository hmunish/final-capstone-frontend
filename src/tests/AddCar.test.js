import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import AddCar from '../components/car/AddCar';
import store from '../redux/store';

test('Car Component matches snapshots', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <AddCar />
      </Provider>
    </BrowserRouter>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
