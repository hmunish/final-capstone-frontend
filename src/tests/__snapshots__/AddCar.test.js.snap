// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car Component matches snapshots 1`] = `
<section
  className="add-new-car"
>
  <div
    className="green-overlay"
  />
  <form
    className="add-car-form"
    onSubmit={[Function]}
  >
    <h1
      className="delete-title"
    >
      Add a New Car
    </h1>
    <ul
      className="form-ul"
    >
      <li
        className="form-li"
      >
        <label
          htmlFor="name"
        >
          Name:
          <br />
          <input
            id="name"
            maxLength={30}
            name="name"
            placeholder="Car Name"
            required={true}
            type="text"
          />
        </label>
      </li>
      <li
        className="form-li"
      >
        <label
          htmlFor="image"
        >
          Image:
          <br />
          <input
            id="image"
            name="image"
            required={true}
            type="text"
          />
        </label>
      </li>
      <li
        className="form-li"
      >
        <label
          htmlFor="description"
        >
          Description:
          <br />
          <input
            id="description"
            name="description"
            placeholder="Model"
            required={true}
            type="text"
          />
        </label>
      </li>
      <li
        className="form-li"
      >
        <label
          htmlFor="deposit"
        >
          Deposit:
          <br />
          <input
            id="deposit"
            min="0"
            name="deposit"
            placeholder="Deposit"
            required={true}
            type="number"
          />
        </label>
      </li>
      <li
        className="form-li"
      >
        <label
          htmlFor="financeFee"
        >
          Finance Fee:
          <br />
          <input
            id="financeFee"
            min="0"
            name="financeFee"
            placeholder="Finance Fee"
            required={true}
            type="number"
          />
        </label>
      </li>
      <li
        className="form-li"
      >
        <label
          htmlFor="optionToPurchaseFee"
        >
          Option to Purchase Fee:
          <br />
          <input
            id="optionToPurchaseFee"
            min="0"
            name="optionToPurchaseFee"
            placeholder="Option to Purchase Fee"
            required={true}
            type="number"
          />
        </label>
      </li>
      <li
        className="form-li"
      >
        <label
          htmlFor="totalAmountPayable"
        >
          Total Amount Payable:
          <br />
          <input
            id="totalAmountPayable"
            min="0"
            name="totalAmountPayable"
            placeholder="Total Amount Payable"
            required={true}
            type="number"
          />
        </label>
      </li>
      <li
        className="form-li"
      >
        <label
          htmlFor="Duration"
        >
          Duration:
          <br />
          <input
            id="duration"
            min="0"
            name="duration"
            placeholder="Duration (Months)"
            required={true}
            type="number"
          />
        </label>
      </li>
      <input
        type="submit"
        value="Add Car"
      />
    </ul>
  </form>
</section>
`;
